src/nvim/ex_getln.c

```diff
@@ -1968,6 +2147,8 @@ static int command_line_changed(CommandLineState *s)
     State = (State & ~CMDPREVIEW);
     close_preview_windows();
     update_screen(SOME_VALID);  // Clear 'inccommand' preview.
+  } else {
+    may_do_incsearch_highlighting(s->firstc, s->count, &s->is_state);
```

Comment:
Does `incsearch` interfere with `inccomand`?

Reply:
They uses different kind of highlighting and kind of interfere with each other. I tried by doing this at first:
```
diff --git a/src/nvim/ex_getln.c b/src/nvim/ex_getln.c
index c58c9b172..bec12ab3b 100644
--- a/src/nvim/ex_getln.c
+++ b/src/nvim/ex_getln.c
@@ -2116,6 +2116,7 @@ static int command_line_changed(CommandLineState *s)
     }
   }
 
+  may_do_incsearch_highlighting(s->firstc, s->count, &s->is_state);
   // 'incsearch' highlighting.
   if (s->firstc == ':'
       && current_sctx.sc_sid == 0    // only if interactive
@@ -2147,8 +2148,6 @@ static int command_line_changed(CommandLineState *s)
     State = (State & ~CMDPREVIEW);
     close_preview_windows();
     update_screen(SOME_VALID);  // Clear 'inccommand' preview.
-  } else {
-    may_do_incsearch_highlighting(s->firstc, s->count, &s->is_state);
   }
 
   if (cmdmsg_rl || (p_arshape && !p_tbidi && enc_utf8)) {
```
If both run at the same time, `may_do_incsearch_highlighting` make the first match's `reverse` attribute set to true, and it fails some of the existing tests. I also tried with having a return value returned by `may_do_incsearch_highlighting`, so that only when it fails we do the `inccommand` block without success.

It also fails some other tests like `:substitute, inccommand=split deactivates if 'redrawtime' is exceeded #5602`, but I can't figure out why.

I figure that since before patch 8.1.0271 `incsearch` doesn't work with `:s`, that means whenever `inccommand` was set, we never uses the highlighting from `incsearch` on `:s` and instead we use the highlighting from `inccommand` block. So I thought we can keep current behaviour by not using `incsearch` highlighting on `:s` and use `inccommand` highlighting like we did before.